compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/SymbolFactory.java

```diff
@@ -133,9 +133,9 @@ public Symbol getBCompiledSymbol(BSymbol symbol, String name) {
                     if (Symbols.isFlagOn(symbol.flags, Flags.RESOURCE)) {
                         return createResourceMethodSymbol((BInvokableSymbol) symbol);
                     }
-                    return createMethodSymbol((BInvokableSymbol) symbol);
+                    return createMethodSymbol((BInvokableSymbol) symbol, name);
                 }
-                return createFunctionSymbol((BInvokableSymbol) symbol, name);
+                return createFunctionSymbol((BInvokableSymbol) symbol, symbol.getOriginalName().getValue());
```

Comment:
Why do we need a different approach just for this?

Reply:
This is a mistake, I'll send the change